src/cts/src/Clock.h

```diff
@@ -218,6 +223,44 @@ class Clock
     return subNets_.back();
   }
 
+  void removeSubNet(ClockSubNet sub_net)
+  {
+    std::set<ClockInst*> sinksToRemove;
+    sinksToRemove.insert(sub_net.getDriver());
+    for (auto it = subNets_.begin(); it != subNets_.end();) {
```

Comment:
What is the cost of iterating over every subnet? Wouldn't the runtime of it be too large?

Reply:
The run time is O(n) for the number of subNets, I don't think there is a better runtime for removing this an element from a std::deque. Also this code shouldn't be run very often, it is to fix some corner cases. But if the runtime becomes a problem we can maintain the subNet deletion at the end of CTS that would not affect the runtime.